tests/profitbricks/models/compute/compute_tests.rb

```diff
@@ -7,59 +7,538 @@
-	  	datacenter = compute.datacenters.create(:name => 'fog-demo',
-                                        		:location => 'de/fra',
-                                       			:description => 'testing fog rest implementation')
+	  		datacenter = compute.datacenters.create(:name => 'fog-demo',
+                                        			:location => 'de/fra',
+                                       				:description => 'testing fog rest implementation')
 			datacenter.wait_for { ready? }
 
 			@datacenter_id = datacenter.id
 
 			datacenter.name == 'fog-demo'
 			datacenter.location == 'de/fra'
 			datacenter.description == 'testing fog rest implementation'
-    end
+	    end
 
-    tests('should retrieve a datacenter by id').succeeds do
-	  	datacenter = compute.datacenters.get(@datacenter_id)
+	    tests('should retrieve a datacenter by id').succeeds do
+		  	datacenter = compute.datacenters.get(@datacenter_id)
 
-	  	datacenter.id != nil
-    end
+		  	datacenter.name == 'fog-demo'
+			datacenter.location == 'de/fra'
+			datacenter.description == 'testing fog rest implementation'
+	    end
 
-    tests('should update a datacenter').succeeds do
-	  	datacenter = compute.datacenters.get(@datacenter_id)
-	  	datacenter.name = datacenter.name + ' - updated'
-	  	datacenter.description = datacenter.description + ' - updated'
-	  	datacenter.update
+	    tests('should update a datacenter').succeeds do
+		  	datacenter = compute.datacenters.get(@datacenter_id)
+		  	datacenter.name = datacenter.name + ' - updated'
+		  	datacenter.description = datacenter.description + ' - updated'
+		  	datacenter.update
 
-	  	datacenter.wait_for { ready? }
+		  	datacenter.wait_for { ready? }
 
-	  	datacenter.name = 'fog-demo - updated'
-	  	datacenter.description = 'testing fog rest implementation - updated'
-    end
+		  	datacenter.name == 'fog-demo - updated'
+		  	datacenter.description == 'testing fog rest implementation - updated'
+	    end
 
 		tests('should retrieve all datacenters').succeeds do
-	  	datacenters = compute.datacenters.all
+		  	datacenters = compute.datacenters.all
+
+		  	datacenters.length > 0
+	    end
+
+	    tests('should retrieve all locations').succeeds do
+		  	locations = compute.locations.all
+
+		  	locations.length > 0
+	    end
+
+	    tests('should retrieve a location by id').succeeds do
+		  	location = compute.locations.get('us/las')
+
+		  	location.name == 'lasvegas'
+	    end
+
+	    tests('should create a volume').succeeds do
+	  		volume = compute.volumes.create(:datacenter_id => @datacenter_id,
+	  										:name => 'fog-demo-volume',
+                                        	:size => 5,
+                                       		:licence_type => 'OTHER',
+                                       		:type => 'HDD')
+			volume.wait_for { ready? }
+
+			@volume_id = volume.id
+
+			volume.name 		== 'fog-demo-volume'
+			volume.size 		== 5
+			volume.type			== 'HDD'
+			volume.licence_type	== 'OTHER'
+	    end
```

Comment:
It was changed to 2 spaces in the latest commit.

Thank you for your feedback.